libkbfs/folder_block_ops.go

```diff
@@ -2551,12 +2566,14 @@ func (fbo *folderBlockOps) SearchForNodes(ctx context.Context,
 		return nodeMap, nil
 	}
 
-	// Start with the root node
 	rootPtr := md.data.Dir.BlockPointer
 	var node Node
 	if cache == fbo.nodeCache {
```

Comment:
Yeah I wrote this comment before I realized why the caches were being passed in. Your explanation here, however, is fantastic, and it might be a good idea to store it somewhere for posterity. The key thing to notice is that `fbo.nodeCache` is the _latest_ view, but the search needs a view fixed in time.